app/javascript/shared/components/MediaQuery.jsx

```diff
@@ -0,0 +1,42 @@
+import PropTypes from 'prop-types';
+import { useMediaQuery } from './useMediaQuery';
+
+/**
+ * A component for evaluating whether or not a CSS media query is matched or not.
+ *
+ * @param {object} props
+ * @param {string} props.query The media query to run.
+ * @param {function} props.render A render prop for using the result of the media query.
+ *
+ * @return {JSX.Element} Runs the render prop function to generate a JSX element
+ *
+ * @example
+ * import { MediaQuery } from '@components/MediaQuery';
+ *
+ * <MediaQuery
+ *   query={`(width >= ${BREAKPOINTS.Medium}px)`}
+ *   render={(matches) => {
+ *     return (
+ *       matches && (
+ *         <aside className="crayons-layout__sidebar-left">
+ *           <TagList
+ *             availableTags={availableTags}
+ *             selectedTag={selectedTag}
+ *             onSelectTag={this.toggleTag}
+ *           />
+ *         </aside>
+ *       )
+ *     );
+ *   }}
+ * />
+ */
+export function MediaQuery({ query, render }) {
+  const matchesBreakpoint = useMediaQuery(query);
+
+  return render(matchesBreakpoint);
```

Comment:
Is the intention of using the `render` callback prop so that the caller can do something like

```
render={(matches) => matches ? <p>I matched</p> : <p> I didn't match </p>}
```

As in, render an alternative component if false? 

Otherwise I'm thinking it would be nice for the component to render `children` if it matches or `null` if not, so you could use it like:

```
<MediaQuery query={`(width >= ${BREAKPOINTS.Medium}px)`}>
  <p>I matched</p>
</MediaQuery>
```

Just a thought / feel free to ignore

Reply:
I had originally done it the way you're suggesting, but the render prop makes it more flexible. Another option could be to do both, i.e. the render prop is optional.

The render prop allows you to use it more like the custom hook.